packages/tools/devtools/devtools-view/src/components/data-visualization/TreeItem.tsx

```diff
@@ -27,16 +30,20 @@ export type TreeItemProps = React.PropsWithChildren<{
  *
  * Intended to be used inside an outer {@link @fluentui/react-components/unstable#Tree} context.
  */
-export function TreeItem(props: TreeItemProps): React.ReactElement {
+export function RecursiveTreeItem(props: TreeItemProps): React.ReactElement {
 	const { children, header } = props;
-
-	const isLeaf = React.Children.count(children) === 0;
+	const { level } = useSubtreeContext_unstable();
+	const open = useTreeItemContext_unstable((ctx) => ctx.open || level === 1);
 
 	return (
-		<FluentTreeItem leaf={isLeaf} data-testid="tree-button">
+		<FluentTreeItem value={level} itemType="branch" style={{ [treeItemLevelToken]: level }}>
 			<FluentTreeItemLayout>{header}</FluentTreeItemLayout>
 
-			<FluentTree>{children}</FluentTree>
+			{open && (
+				<FluentTree>
+					<RecursiveTreeItem header={header}> {children} </RecursiveTreeItem>
```

Comment:
I am not sure if passing in the same `header: React.ReactElement` is correct. Trying to figure out a way to construct new component.